src/runtime-rs/crates/runtimes/src/manager.rs

```diff
@@ -276,6 +283,13 @@ impl RuntimeHandlerManager {
                     .start_process(&process_id)
                     .await
                     .context("start process")?;
+
+                // update sandbox resource in case when starting a paused container
+                instance
+                    .update_sandbox_resource()
```

Comment:
I think this is no need to `update_sandbox_resource()` when starting the process, the CPU and memory resources are updated when creating or updating the container.

Reply:
This is for the case like
```
1. stop a container (the resource will not be deallocated)
2. start a container (the resource previously allocated is enough, so this start will not allocate any resource)
```

Reply:
@Ji-Xinyou As you say for case 2 

> This is for the case like
> 
> ```
> 1. stop a container (the resource will not be deallocated)
> 2. start a container (the resource previously allocated is enough, so this start will not allocate any resource)
> ```
@Ji-Xinyou As you say in case 2 "the resource previously allocated is enough, so this start will not allocate any resource", so I think no need to call `update_sandbox_resource()`.

Reply:
> @Ji-Xinyou As you say for case 2
> 
> > This is for the case like
> > ```
> > 1. stop a container (the resource will not be deallocated)
> > 2. start a container (the resource previously allocated is enough, so this start will not allocate any resource)
> > ```
> 
> @Ji-Xinyou As you say in case 2 "the resource previously allocated is enough, so this start will not allocate any resource", so I think no need to call `update_sandbox_resource()`.

I remember this was the 2.0 behavior.
I guess this was for those VMMs who support hot-unplug. To handle your case, we could only update the resource for such VMM, wdyt?

Reply:
Currently, the [`cpus_required()`](https://github.com/openanolis/kata-containers/blob/runtime_vcpu_resize/src/runtime-rs/crates/runtimes/virt_container/src/container_manager/manager.rs#L293) calculate all to container CPU without check the container state, so the CPU number only changes for `Create/Update/Delete Container` not the `Start Container`. If you want to include the `Pause State` of a container, the `cpus_required()` should consider the container state too, but I wonder if this case is needed.